### ISSUE TITLE
Incorrect program code

### ISSUE THREAD

**Issue by Snowbelldog** [Auties00/Whatsapp4j#210]
First of all, thank you for this great project and the strong work.

Incorrect program code in the class `it.auties.whatsapp.model.message.model.MediaMessage` in the two methods named save.

Here the code of the two methods
```
  /**
   * Saves this media to the internal memory of the host. Throws an error if the media cannot be
   * downloaded successfully.
   *
   * @return the non-null path where the file was downloaded
   */
  public Path save() {
    return LocalFileSystem.of("medias")
        .resolve("%s.%s".formatted(Bytes.ofRandom(5)
            .toHex(), mediaType().fileExtension()));
  }

  /**
   * Saves this media to the provided path. Throws an error if the media cannot be downloaded
   * successfully.
   *
   * @param path the non-null path where the media should be written.
   * @return the non-null path where the file was downloaded
   */
  public Path save(@NonNull Path path) {
    var result = decodedMedia();
    Validate.isTrue(result.isEmpty(), "Cannot save media");
    try {
      Files.createDirectories(path.getParent());
      Files.write(path, result.get(), StandardOpenOption.CREATE,
          StandardOpenOption.TRUNCATE_EXISTING);
    } catch (IOException exception) {
      throw new UncheckedIOException("Cannot write media to file", exception);
    }
    return path;
  }
```

In the first method it is described that the path is generated and the file is downloaded afterwards. However, this does not happen and only the path is generated.

In the actual save method the check is Validate.isTrue but the boolean must be true otherwise there is the error. So here simply result.isPresent() would be correct.

Otherwise a very great project. I hope that the Java version will be changed in the future so that it also runs without the preview version.

**Comment by Auties00** [comment]
> First of all, thank you for this great project and the strong work.
> 
> Incorrect program code in the class `it.auties.whatsapp.model.message.model.MediaMessage` in the two methods named save.
> 
> Here the code of the two methods
> ```
>   /**
>    * Saves this media to the internal memory of the host. Throws an error if the media cannot be
>    * downloaded successfully.
>    *
>    * @return the non-null path where the file was downloaded
>    */
>   public Path save() {
>     return LocalFileSystem.of("medias")
>         .resolve("%s.%s".formatted(Bytes.ofRandom(5)
>             .toHex(), mediaType().fileExtension()));
>   }
> 
>   /**
>    * Saves this media to the provided path. Throws an error if the media cannot be downloaded
>    * successfully.
>    *
>    * @param path the non-null path where the media should be written.
>    * @return the non-null path where the file was downloaded
>    */
>   public Path save(@NonNull Path path) {
>     var result = decodedMedia();
>     Validate.isTrue(result.isEmpty(), "Cannot save media");
>     try {
>       Files.createDirectories(path.getParent());
>       Files.write(path, result.get(), StandardOpenOption.CREATE,
>           StandardOpenOption.TRUNCATE_EXISTING);
>     } catch (IOException exception) {
>       throw new UncheckedIOException("Cannot write media to file", exception);
>     }
>     return path;
>   }
> ```
> 
> In the first method it is described that the path is generated and the file is downloaded afterwards. However, this does not happen and only the path is generated.
> 
> In the actual save method the check is Validate.isTrue but the boolean must be true otherwise there is the error. So here simply result.isPresent() would be correct.
> 
> Otherwise a very great project. I hope that the Java version will be changed in the future so that it also runs without the preview version.

Thanks, will be fixed in the next release

**Comment by Auties00** [comment]
Fixed on master branch, will be shipped in release 3.1 this night

**Comment by Auties00** [comment]
Released 3.1
